tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc

```diff
@@ -567,59 +539,32 @@ static std::vector<string> GetROCDLPaths(int amdgpu_version,
 // Emits the given module to HSA Code Object. target_machine is an initialized
 // TargetMachine for the AMDGPU target.
 StatusOr<std::vector<uint8>> EmitModuleToHsaco(
-    Module* module, llvm::TargetMachine* target_machine) {
-  auto* env = tensorflow::Env::Default();
-  std::vector<std::string> tempdir_vector;
-  env->GetLocalTempDirectories(&tempdir_vector);
-  if (tempdir_vector.empty()) {
-    return xla::InternalError(
-        "Unable to locate a temporary directory for compile-time artifacts.");
-  }
-  std::string tempdir_name = tempdir_vector.front();
-  VLOG(1) << "Compile-time artifacts located at: " << tempdir_name;
-
-  // Prepare filenames for all stages of compilation:
-  // IR, binary ISA, and HSACO.
-  std::string ir_filename = absl::StrCat(module->getModuleIdentifier(), ".ll");
-  std::string ir_path = tensorflow::io::JoinPath(tempdir_name, ir_filename);
-
-  std::string isabin_filename =
-      absl::StrCat(module->getModuleIdentifier(), ".o");
-  std::string isabin_path =
-      tensorflow::io::JoinPath(tempdir_name, isabin_filename);
-
-  std::string hsaco_filename =
-      absl::StrCat(module->getModuleIdentifier(), ".hsaco");
-  std::string hsaco_path =
-      tensorflow::io::JoinPath(tempdir_name, hsaco_filename);
+    Module* module, llvm::TargetMachine* target_machine, const string& optimized_ir_path, const string& isabin_path, const string& hsaco_path) {
+  std::string error_message;
 
-  std::error_code ec;
+  // Locate llc.
+  std::string llc_path = tensorflow::io::JoinPath("/opt/rocm", "llvm/bin");
+  auto llc_program = llvm::sys::findProgramByName("llc", {llc_path});
+  if (!llc_program) {
+    return xla::InternalError("unable to find llc in PATH: %s", llc_program.getError().message());
+  }
+  std::vector<llvm::StringRef> llc_args{
+      llvm_ir::AsStringRef("llc"),
+      llvm_ir::AsStringRef("-march=amdgcn"),
+      llvm_ir::AsStringRef("-mcpu=gfx942"),
```

Comment:
hi, where is -mcpu is defined and used ? (MI30X, gfx942 arch).

Reply:
Please refer to: https://llvm.org/docs/CommandGuide/llc.html#end-user-options